src/pandas_profiling/report/structure/variables/render_timeseries.py

```diff
@@ -239,11 +235,22 @@ def render_timeseries(config: Settings, summary: dict) -> dict:
         sequence_type="grid",
     )
 
+    if isinstance(summary["histogram"], list):
+        hist_data = histogram(
+            config,
+            [x[0] for x in summary["histogram"]],
+            [x[1] for x in summary["histogram"]],
+        )
+        hist_caption = f"<strong>Histogram with fixed size bins</strong> (bins={len(summary['histogram'][0][1]) - 1})"
+    else:
+        hist_data = histogram(config, *summary["histogram"])
+        hist_caption = f"<strong>Histogram with fixed size bins</strong> (bins={len(summary['histogram'][1]) - 1})"
+
     hist = Image(
-        histogram(config, *summary["histogram"]),
+        hist_data,
         image_format=image_format,
         alt="Histogram",
-        caption=f"<strong>Histogram with fixed size bins</strong> (bins={len(summary['histogram'][1]) - 1})",
+        caption=hist_caption,
```

Comment:
Feels like the `len` should be read from the freshly-computed `hist_data` object instead of slightly differently in the two branches.

Reply:
hist_data is a string with xml of the histogram